src/lib/utils/thorchain/hooks/useThorchainFromAddress.tsx

```diff
@@ -0,0 +1,51 @@
+import type { AccountId, AssetId } from '@shapeshiftoss/caip'
+import type { HDWallet } from '@shapeshiftoss/hdwallet-core'
+import type { AccountMetadata } from '@shapeshiftoss/types'
+import { useQuery } from '@tanstack/react-query'
+import { reactQueries } from 'react-queries'
+import type { getThorchainLpPosition } from 'pages/ThorChainLP/queries/queries'
+import type { getThorchainSaversPosition } from 'state/slices/opportunitiesSlice/resolvers/thorchainsavers/utils'
+
+import type { getThorchainLendingPosition } from '../lending'
+
+type UseThorchainFromAddressArgs = {
+  accountId: AccountId | undefined
+  assetId: AssetId | undefined
+  opportunityId?: string | undefined
```

Comment:
I feel like we should pick one or the other here. In this case, it would be beneficial to force the consumer to specify `opportunityId` regardless instead of being able to omit imo
```suggestion
  opportunityId: string | undefined
```

Reply:
fa21fd1d0b